apps/festival/festival/src/app/dashboard/home/home.component.ts

```diff
@@ -70,6 +73,27 @@ export class HomeComponent {
     map(analytics => analytics.filter(analytic => analytic.name === 'pageView'))
   );
 
+  activeBuyers$ = this.titleAnalytics$.pipe(
+    filter(analytics => analytics.length > 0),
+    map(analytics => {
+      const uids = unique(analytics.map(analytic => analytic.meta.uid));
+      const orgIds = unique(analytics.map(analytic => analytic.meta.orgId));
+      return { uids, orgIds, analytics };
+    }),
+    joinWith({
+      users: ({ uids }) => this.userService.valueChanges(uids),
+      orgs: ({ orgIds }) => this.orgService.valueChanges(orgIds)
+    }, { shouldAwait: true }),
+    map(({ users, orgs, analytics }) => {
+      const aggregatedUsers = users.map(user => {
+        const org = orgs.find(o => o.id === user.orgId);
+        const analyticsOfUser = analytics.filter(analytic => analytic.meta.uid === user.uid);
+        return aggregate(analyticsOfUser, { user, org });
+      });
+      return aggregatedUsers.sort((userA, userB) => userA.countEvents - userB.countEvents)
+    })
```

Comment:
```suggestion
    map(({ users, orgs, analytics }) => {
      return users.map(user => {
        const org = orgs.find(o => o.id === user.orgId);
        const analyticsOfUser = analytics.filter(analytic => analytic.meta.uid === user.uid);
        return aggregate(analyticsOfUser, { user, org });
      });
    }),
    map(aggregated => aggregated.sort((a, b) => a.total - b.total))
```

This way each `map` has a dedicated purpose